contracts/orderbook/Logic/OrderBookOrderPlace.sol

```diff
@@ -0,0 +1,258 @@
+        );
+        require(
+            Order.loanID != 0
+                ? _caseChecks(Order.loanID, Order.base, Order.loanTokenAddress)
+                : true,
+            "OrderBook: cases not passed"
+        );
+        require(
+            Order.orderType == IOrderBook.OrderType.LIMIT_OPEN
+                ? protocol.isLoanPool(Order.iToken)
+                : true,
+            "OrderBook: invalid iToken specified"
+        );
+        require(
+            Order.orderType == IOrderBook.OrderType.LIMIT_OPEN
+                ? Order.loanID == 0 ||
+                    _isActiveLoan(Order.loanID)
+                : _isActiveLoan(Order.loanID),
+            "OrderBook: inactive loan"
+        );
+        (uint256 amountUsed, address usedToken) = Order.loanTokenAmount > Order.collateralTokenAmount
+            ? (Order.loanTokenAmount, Order.loanTokenAddress)
+            : (Order.collateralTokenAmount, Order.base);
+        require(
+            queryRateReturn(usedToken, USDC, amountUsed) >=
+                MIN_AMOUNT_IN_USDC,
+            "OrderBook: Order too small"
+        );
+        require(Order.trader == msg.sender, "OrderBook: invalid trader");
+        require(!Order.isCancelled, "OrderBook: invalid order state");
+        mainOBID++;
+        bytes32 ID = keccak256(abi.encode(msg.sender, mainOBID));
+        require(IDeposits(vault).getTokenUsed(ID) == address(0), "Orderbook: collision"); //in the very unlikely chance of collision on ID error is thrown
+        _allOrders[ID] = Order;
+        _allOrders[ID].orderID = ID;
+        _histOrders[msg.sender].add(ID);
+        _allOrderIDs.add(ID);
+        if (Order.orderType == IOrderBook.OrderType.LIMIT_OPEN) {
+            IDeposits(vault).deposit(ID, amountUsed, msg.sender, usedToken);
+        }
+        emit OrderPlaced(
+            msg.sender,
+            Order.orderType,
+            Order.amountReceived,
+            ID,
+            Order.base,
+            Order.loanTokenAddress
+        );
+    }
+
+    function amendOrder(IOrderBook.Order calldata Order) external pausable {
+        require(Order.loanDataBytes.length < 3500, "OrderBook: loanDataBytes too complex");
+        require(
+            _abs(
+                int256(Order.loanTokenAmount) -
+                    int256(Order.collateralTokenAmount)
+            ) == int256(Order.loanTokenAmount + Order.collateralTokenAmount),
+            "OrderBook: only one token can be used"
+        );
+        require(
+            Order.base == _allOrders[Order.orderID].base &&
+                Order.loanTokenAddress ==
+                _allOrders[Order.orderID].loanTokenAddress,
+            "OrderBook: invalid tokens"
+        );
+        /*require(
+            Order.price > swapRate
+                ? (Order.price - swapRate) < (Order.price * 25) / 100
+                : (swapRate - Order.price) < (swapRate * 25) / 100,
+            "price too far away"
+        );*/
+        require(
```

Comment:
some of those checks are repeating. maybe woth having 
`verifySomething` type of functions?

Reply:
yes